torch/csrc/distributed/c10d/Collectives.hpp

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <ATen/core/ivalue.h>
+#include <chrono>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include <c10/macros/Macros.h>
+#include <torch/custom_class.h>
+
+namespace c10d {
+
+using namespace std::chrono_literals;
+
+class TORCH_API Collectives : public torch::CustomClassHolder {
+ public:
+  virtual void barrier(
+      const std::string& prefix,
```

Comment:
Is 'prefix' specific to store based implementation of the APIs? If so, I imagine the APIs could be implemented with non-store semantics

Reply:
It refers to the store based APIs but other implementations have a similar concept of an operation ID. I.e. the label field in `https://fburl.com/code/dq8shbwi`.

It may be best to change this name from prefix to "operation_uid"

Reply:
I think this is still a better fit for label -- all operations on store must be sequential regardless of prefix.

If you want a fully separate namespace you can use a different PrefixStore and create a new StoreCollectives instance -- similar to how you would use a different sub_world_id and create a new SCCollectives instance

Reply:
Renamed to generic `key`